src/test/regress/expected/single_shard_table_udfs.out

```diff
@@ -0,0 +1,219 @@
+ citus_table_is_visible | relation_is_a_known_shard
+---------------------------------------------------------------------
+ t                      | f
+(1 row)
+
+-- should fail, maybe support in the future
+SELECT create_reference_table('null_dist_key_table');
+ERROR:  table "null_dist_key_table" is already distributed
+SELECT create_distributed_table('null_dist_key_table', 'a');
+ERROR:  table "null_dist_key_table" is already distributed
+SELECT create_distributed_table_concurrently('null_dist_key_table', 'a');
+ERROR:  table "null_dist_key_table" is already distributed
+SELECT citus_add_local_table_to_metadata('null_dist_key_table');
+ERROR:  table "null_dist_key_table" is already distributed
+-- test altering distribution column, fails for single shard tables
+SELECT alter_distributed_table('null_dist_key_table', distribution_column := 'a');
+ERROR:  relation null_dist_key_table should be a hash distributed table
+-- test altering shard count, fails for single shard tables
+SELECT alter_distributed_table('null_dist_key_table', shard_count := 6);
+ERROR:  relation null_dist_key_table should be a hash distributed table
+-- test shard splitting udf, fails for single shard tables
+SELECT nodeid AS worker_1_node FROM pg_dist_node WHERE nodeport=:worker_1_port \gset
+SELECT nodeid AS worker_2_node FROM pg_dist_node WHERE nodeport=:worker_2_port \gset
+SELECT citus_split_shard_by_split_points(
+	1720000,
+	ARRAY['-1073741826'],
+	ARRAY[:worker_1_node, :worker_2_node],
+    'block_writes');
+ERROR:  Cannot split shard as operation is only supported for hash distributed tables.
+-- test alter_table_set_access_method
+SELECT alter_table_set_access_method('null_dist_key_table', 'columnar');
+NOTICE:  creating a new table for null_dist_key_udfs.null_dist_key_table
+NOTICE:  moving the data of null_dist_key_udfs.null_dist_key_table
+NOTICE:  dropping the old null_dist_key_udfs.null_dist_key_table
+NOTICE:  renaming the new table to null_dist_key_udfs.null_dist_key_table
+ alter_table_set_access_method
+---------------------------------------------------------------------
+
+(1 row)
+
+-- undistribute
+SELECT undistribute_table('null_dist_key_table');
+NOTICE:  creating a new table for null_dist_key_udfs.null_dist_key_table
+NOTICE:  moving the data of null_dist_key_udfs.null_dist_key_table
+NOTICE:  dropping the old null_dist_key_udfs.null_dist_key_table
+NOTICE:  renaming the new table to null_dist_key_udfs.null_dist_key_table
+ undistribute_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- verify that the metadata is gone
+SELECT COUNT(*) = 0 FROM pg_dist_partition;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT COUNT(*) = 0 FROM pg_dist_placement;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT COUNT(*) = 0 FROM pg_dist_shard;
```

Comment:
Can we add some filters for the table we're looking for? Above assumptions would hold based on where we're placing this test in the schedule file but this may not be the case in future.

Reply:
done